turludock/generate_nvidia_templated_files.py

```diff
@@ -0,0 +1,201 @@
+import importlib.resources
+from string import Template
+
+from loguru import logger
+
+from turludock.config_sanity import is_cuda_cudnn_version_combination_supported, is_cuda_version_supported
+from turludock.yaml_load import load_cuda_config, load_cudnn_config
+
+
+def generate_cuda_base(cuda_version: str, ubuntu_version: str) -> str:
```

Comment:
It is hard for me to see the differences between the different functions here, but at first glance it seems there is a lot of shared code in these functions. You might want to move the shared code to a dedicated function in order to reduce the duplication of code.

Reply:
Are you talking about `load_cuda_config` and `load_cudnn_config`?